GoogleDataTransport/Tests/Lifecycle/GDTLifecycleTest.m

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright 2018 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import <XCTest/XCTest.h>
+
+#import <GoogleDataTransport/GoogleDataTransport.h>
+
+#import "GDTStorage_Private.h"
+#import "GDTTransformer_Private.h"
+#import "GDTUploadCoordinator_Private.h"
+
+#import "GDTLifecycleTestPrioritizer.h"
+#import "GDTLifecycleTestUploader.h"
+#import "GDTStorage+Testing.h"
+#import "GDTUploadCoordinator+Testing.h"
+
+/** Waits for the result of waitBlock to be YES, or times out and fails.
+ *
+ * @param waitBlock The block to periodically execute.
+ * @param timeInterval The timeout.
+ */
+#define GDTWaitForBlock(waitBlock, timeInterval)                                                  \
```

Comment:
Why a macro instead of a function?

Reply:
If I use the function, the failure will appear to be in the function and Xcode's tools won't make it obvious where in the actual test the failure occurred. Also, the standard XCT macros require self, which'll be missing in a function. The macro tells me exactly where the failure was, which is important because I'd call the function/macro multiple times in a single test.